go/deploy.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"text/template"
+
+	"github.com/bjartek/overflow/overflow"
+	"github.com/onflow/cadence"
+)
+
+type OverflowUtils struct {
+	O *overflow.OverflowState
+}
+
+type Addresses struct {
+	FungibleToken   string
+	OnChainMultiSig string
+}
+
+func main() {
+	otu := DeployOverflow()
+	otu.DeployFiatTokenContract("treasuryOwner", "USDC", "0.1.0")
```

Comment:
could you help me understand why USDC require special script to deploy? and is this script only does deploy or there are other setups we are trying to achieve?

Reply:
USDC init method has a lot of parameters, and it cannot be deployed over flow.json, so the script needed to be added.
I took this and modified it in order to work https://github.com/flow-usdc/flow-usdc/blob/main/lib/go/deploy/USDC.go
This script does the deployment of all contracts and the USDC contract too, and it's only for local testing, because on testnet this is already deployed (it can be seen on which address in the flow.json)

Reply:
i see. So it's meant to achieve what `deploy.js` does too.  Maybe i missed it but how does it deploy all contracts? which part of code does that?

Reply:
Overflow library does it with this part of the code:
`&OverflowUtils{O: overflow.NewOverflowEmulator().Start()}`
it reads the flow.json and deploys all the contracts from there, and then we call the DeployFiatTokenContract function to deploy the FiatToken.cdc